apps/be-api/src/app/routes/products.ts

```diff
@@ -0,0 +1,57 @@
+import { FastifyInstance } from "fastify";
+import { Product } from "../models/product.models";
+
+export default async function productRoutes(fastify: FastifyInstance) {
+  fastify.get("/products", async (_, reply) => {
+    const products = await Product.find();
+    return reply.send(products);
+  });
+
+  fastify.get("/products/:id", async (request, reply) => {
+    try {
+      const product = await Product.findById((request.params as any).id);
+      if (!product) return reply.status(404).send({ message: "Not Found" });
+      return reply.send(product);
+    } catch (error) {
+      return reply.status(500).send(error);
+    }
+  });
```

Comment:
https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/
read this article, the best practice is to validate params and body everywhere